tests/i18n/test_extraction.py

```diff
@@ -687,6 +687,31 @@ def test_all_locales(self):
         self.assertTrue(os.path.exists(self.PO_FILE_KO))
         self.assertFalse(os.path.exists('locale/_do_not_pick/LC_MESSAGES/django.po'))
 
+    def test_locales_format_valid(self):
+        """Check the standard output(stdout) message in case the format of locale is valid"""
+        out = StringIO()
+        LOCALE = ('en-IN', 'th_TH_TH', 'asa_TZ', 'az_Cyrl', 'yue_Hant_HK', 'haw_US')
+        management.call_command("makemessages", locale=LOCALE, verbosity=1, stdout=out)
+        self.assertNotIn("invalid", out.getvalue())
```

Comment:
Do you mean creating a separate class for validity tests? If yes, then I may do that as well. Please confirm.

Reply:
or is it something like ` with self.subTest(test_content):`. Reference [https://github.com/django/django/commit/e065b293878b1e3ea56655aa9d33e87576cd77ff](https://github.com/django/django/commit/e065b293878b1e3ea56655aa9d33e87576cd77ff)

Reply:
```
for locale in locales:
    with self.subTest(locale):
        #  call makemessages with that single locale
        self.assertWhatever(...
```
The advantage is that the individual subtests will report failures individually.

Reply:
Thanks a lot, @jacobtylerwalls. The suggestion seems great and the idea of subTest would be good to implement.

Reply:
May you please help me with why the test `pull-requests-windows/database=sqlite3,label=windows,python=Python39` is failed on this PR so that I may solve that as well. I am unable to find any solution to this yet :).